test/NuGet.Core.Tests/NuGet.Packaging.Test/RelatedFileExtensionPropertyTests.cs

```diff
@@ -0,0 +1,197 @@
+        {
+            // Arrange
+            var collection = new ContentItemCollection();
+            string assembly = "lib/net50/system.dll";
+            string[] paths = new string[]
+            {
+                "lib/net50/system.dll",
+                "lib/net50/system.exe",
+                "lib/net50/system.Core.dll"
+            };
+            collection.Load(paths);
+
+            // Act
+            string relatedProperty = collection.GetRelatedFileExtensionProperty(assembly, CreateAssetsFromPathList(paths));
+
+            // Assert
+            Assert.Equal(null, relatedProperty);
+        }
+
+        [Fact]
+        public void GetRelatedFileExtensionProperty_MultipleAssemblyAssetsWithVariousPrefixMixedCases_GetNullProperty()
+        {
+            // Arrange
+            var collection = new ContentItemCollection();
+            string assembly = "lib/net50/system.dll";
+            string[] paths = new string[]
+            {
+                "lib/net50/system.dll",
+                "lib/net50/system.EXE",
+                "lib/net50/system.Core.DLL",
+                "lib/net50/system.NET.DLL",
+            };
+            collection.Load(paths);
+
+            // Act
+            string relatedProperty = collection.GetRelatedFileExtensionProperty(assembly, CreateAssetsFromPathList(paths));
+
+            // Assert
+            Assert.Equal(null, relatedProperty);
+        }
+
+        [Fact]
+        public void GetRelatedFileExtensionProperty_MultipleRelatedFilesWithSamePrefix_GetCorrectProperty()
+        {
+            // Arrange
+            var collection = new ContentItemCollection();
+            string assembly = "lib/net50/system.test.dll";
+            string[] paths = new string[]
+            {
+                "lib/net50/system.test.dll",
+                "lib/net50/system.test.pdb",
+                "lib/net50/system.test.xml",
+                "lib/net50/system.test.dll.config",
+            };
+            collection.Load(paths);
+
+            // Act
+            string relatedProperty = collection.GetRelatedFileExtensionProperty(assembly, CreateAssetsFromPathList(paths));
+
+            string expectedRelatedProperty = ".dll.config;.pdb;.xml";
+            // Assert
+            Assert.Equal(expectedRelatedProperty, relatedProperty);
+        }
+
+        [Fact]
+        public void GetRelatedFileExtensionProperty_MultipleRelatedFilesWithDifferentPrefix_GetNullProperty()
+        {
+            // Arrange
+            var collection = new ContentItemCollection();
+            string assembly = "lib/net50/system.test.dll";
+            string[] paths = new string[]
+            {
+                "lib/net50/system.test.dll",
+                "lib/net50/system.test2.dll",
+                "lib/net50/system.test2.pdb",
+                "lib/net50/system.test2.xml",
+                "lib/net50/system.test2.dll.config"
```

Comment:
Can we make sure we have a `dll.config` test. That one is a special case given that it has the dll extension but it's supposed to be selected.

Reply:
oh mb :D 
Resolving,